src/app/components/add-task/add-task.component.html

```diff
@@ -23,9 +25,11 @@ <h1 class="dialog-title">Add a new task</h1>
                         {{type}}
                     </option>
                     </select>
+                    <app-form-error [control]="formGroup.controls['type']">
+                    </app-form-error>
                 </div>
                 <div>
-                    <button type="submit" class="button round">Add Task!</button>
+                    <button #submitButton type="submit" class="button round" [ngClass]="{'disabled': !formGroup.valid}">Add Task!</button>
```

Comment:
I think you can use [disabled] input to disable/enable the button [disabled]=!formGroup.valid